### DISCUSSION TITLE
How To Add Include Select Relate Type to useQuery's result.

### DISCUSSION THREAD

**Discussion by cqh963852** [omar-dulaimi/prisma-trpc-generator#61] [Q&A] [answered]
The type of findManyUser's return type is **User[]**
![image](https://user-images.githubusercontent.com/17702287/212950409-f78e99c6-1f28-404a-bb62-0fd2c90d27a4.png)

After passing in the **include** parameter，The type should be changed.

![image](https://user-images.githubusercontent.com/17702287/212950674-013c71ec-96c3-499e-9c12-e068eee007c9.png)

But at `@trpc/react-query` 's useQuery，the type is the return type of `findManyUser`

![image](https://user-images.githubusercontent.com/17702287/212951423-85d2b6cc-8a6b-439b-9ff2-9603f9ff1182.png)

It there any way can make the `useQuery` automatically infer whether a type needs to be added.

![image](https://user-images.githubusercontent.com/17702287/212954910-54500d2d-c9b3-49b6-a8ac-e451be623660.png)

Just like prisma's type

![image](https://user-images.githubusercontent.com/17702287/212951891-7973761c-3578-4316-a81e-c83fcb5cdb86.png)

**Comment by cqh963852** [comment]
![image](https://user-images.githubusercontent.com/17702287/212953371-40b9bdcb-845e-4768-a579-148720d2d39c.png)

The generic type of query will be inferd as `User[]` by the return typeof `query`.

**Comment by cqh963852** [comment, marked as answer]
I found one solution.
```
type SomeType = Prisma.UserGetPayload<typeof query>;
trpc.schema.findManyUser.useQuery<any, SomeType[]>(query)
```

The **SomeType** can be defined by user.

**Reply by omar-dulaimi** [reply]
Hey @cqh963852 

Your workaround does solve the issue for now. It was reported here https://github.com/omar-dulaimi/prisma-trpc-generator/issues/62

Please let me know if you have suggestions for a fix for this

**Reply by cqh963852** [reply]
For a solution using generated code.

Users can customize function encapsulation.

Just like

```
const useFindManyUser = <Q extends SomeQueryType>(query: Q):PrismaGetPayLoadType<Q> => {
    // implement
};
```

But to some extent, it will affect the convenience of use.

https://github.com/idealjs/ponder/blob/main/packages/generator-browser/src/templates/genSWRHelper.ejs

You can check my swr helper template. Also can be used for react-query.

**Reply by just-maik** [reply]
This should be done by the generator. Otherwise its really confusing for new library users to figure out how to type-safely query relational objects.
